### PR TITLE
CHE-233:Rework Project API with new VFS

### PR DESCRIPTION
**Pull request by vparfonov**
Signed-off-by: Vitaly Parfonov vparfonov@codenvy.com

### CODE REVIEW

**Comment by vparfonov** [comment]
@skabashnyuk @gazarenkov @azatsarynnyy @evidolob @vzhukovskii @evoevodin

**Comment by TylerJewell** [comment]
What documentation must be added to plugin docs so that new extension developers can properly use this, or does it have no impact?

**Comment by vparfonov** [comment]
We don't need any updates

File: agent/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/type/ProjectTypeUtils.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * @author Vitalii Parfonov
+ */
+public class ProjectTypeUtils {
+
+    public static ProjectConfigDto ensure(@NotNull ProjectConfigDto projectConfigDto,
```
Comment (skabashnyuk) [inline agent/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/type/ProjectTypeUtils.java]: Ensure what?

File: agent/che-core-api-project/src/main/java/org/eclipse/che/api/project/shared/dto/ItemReference.java [hunk 2]
```diff
@@ -68,12 +65,12 @@
...
-
-    void setCreated(long created);
-
-    ItemReference withCreated(long created);
+//    /** creating date. */
+//    long getCreated();
+//
+//    void setCreated(long created);
+//
+//    ItemReference withCreated(long created);
```
Comment (skabashnyuk) [inline agent/che-core-api-project/src/main/java/org/eclipse/che/api/project/shared/dto/ItemReference.java]: remove if it's unused

File: agent/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/type/ProjectTypeUtils.java (hunk 1 above)
Comment (vparfonov) [inline agent/che-core-api-project/src/main/java/org/eclipse/che/api/project/server/type/ProjectTypeUtils.java]: good question :) need to check

**Comment by vparfonov** [comment]
fixed

**Comment by skabashnyuk** [comment]
ok

**Pull request merged**